### DISCUSSION TITLE
newbie: can't figure out what are the errors in my config.

### DISCUSSION THREAD

**Discussion by juxtapositions-g** [i3/i3#5941] [Q&A] [answered]
I can see it's saying that the syntax is kind of wrong, "expected one of these tokens.." but I don't understand what did I do wrong whilst customizing i3.
Much appreciated!

ERROR: CONFIG: Expected one of these tokens: <end>, '#', 'set ', 'set   ', 'set_from_r
esource', 'include', 'bindsym', 'bindcode', 'bind', 'bar', 'font', 'mode', 'gaps', 'sm
art_borders', 'smart_gaps', 'floating_minimum_size', 'floating_maximum_size', 'floatin
g_modifier', 'default_orientation', 'workspace_layout', 'default_border', 'new_window'
, 'default_floating_border', 'new_float', 'hide_edge_borders', 'for_window', 'assign',
 'no_focus', 'focus_follows_mouse', 'mouse_warping', 'focus_wrapping', 'force_focus_wr
apping', 'force_xinerama', 'force-xinerama', 'disable_randr15', 'disable-randr15', 'wo
rkspace_auto_back_and_forth', 'fake_outputs', 'fake-outputs', 'force_display_urgency_h
int', 'focus_on_window_activation', 'title_align', 'show_marks', 'workspace', 'ipc_soc
ket', 'ipc-socket', 'ipc_kill_timeout', 'restart_state', 'popup_during_fullscreen', 't
iling_drag', 'exec_always', 'exec', 'client.background', 'client.focused_inactive', 'c
lient.focused_tab_title', 'client.focused', 'client.unfocused', 'client.urgent', 'clie
nt.placeholder'
ERROR: CONFIG: (in file /home/qi/.config/i3/config)
ERROR: CONFIG: Line   1: j This file has been auto-generated by i3-config-wizard(1).
ERROR: CONFIG:           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR: CONFIG: Line   2: # It will not be overwritten, so edit it as you like.
ERROR: CONFIG: Line   3: #
ERROR: CONFIG: Expected one of these tokens: <end>, '#', 'set ', 'set   ', 'set_from_r
esource', 'include', 'bindsym', 'bindcode', 'bind', 'bar', 'font', 'mode', 'gaps', 'sm
art_borders', 'smart_gaps', 'floating_minimum_size', 'floating_maximum_size', 'floatin
g_modifier', 'default_orientation', 'workspace_layout', 'default_border', 'new_window'
, 'default_floating_border', 'new_float', 'hide_edge_borders', 'for_window', 'assign',
 'no_focus', 'focus_follows_mouse', 'mouse_warping', 'focus_wrapping', 'force_focus_wr
apping', 'force_xinerama', 'force-xinerama', 'disable_randr15', 'disable-randr15', 'wo
rkspace_auto_back_and_forth', 'fake_outputs', 'fake-outputs', 'force_display_urgency_h
int', 'focus_on_window_activation', 'title_align', 'show_marks', 'workspace', 'ipc_soc
ket', 'ipc-socket', 'ipc_kill_timeout', 'restart_state', 'popup_during_fullscreen', 't
iling_drag', 'exec_always', 'exec', 'client.background', 'client.focused_inactive', 'c
lient.focused_tab_title', 'client.focused', 'client.unfocused', 'client.urgent', 'clie
nt.placeholder'
ERROR: CONFIG: (in file /home/qi/.config/i3/config)
ERROR: CONFIG: Line 233: 
ERROR: CONFIG: Line 234: ### AUTOSTART-TINGS ###
ERROR: CONFIG: Line 235: exec --no-startup-id "feh --bg-fill -z ~/wallpapers" &
ERROR: CONFIG:                                                                ^
ERROR: CONFIG: Line 236: exec_always --no-startup-id "~/.config/polybar/launch.sh" &
ERROR: CONFIG: Line 237: exec --no-startup-id "redshift ~/.config/redshift/redshift.co
nf" &
ERROR: CONFIG: Expected one of these tokens: <end>, '#', 'set ', 'set   ', 'set_from_r
esource', 'include', 'bindsym', 'bindcode', 'bind', 'bar', 'font', 'mode', 'gaps', 'smapping', 'force_xinerama', 'force-xinerama', 'disable_randr15', 'disable-randr15', 'workspace_auto_back_and_forth', 'fake_outputs', 'fake-outputs', 'force_display_urgency_hint', 'focus_on_window_activation', 'title_align', 'show_marks', 'workspace', 'ipc_socket', 'ipc-socket', 'ipc_kill_timeout', 'restart_state', 'popup_during_fullscreen', 'tiling_drag', 'exec_always', 'exec', 'client.background', 'client.focused_inactive', 'client.focused_tab_title', 'client.focused', 'client.unfocused', 'client.urgent', 'client.placeholder'
ERROR: CONFIG: (in file /home/qi/.config/i3/config)
ERROR: CONFIG: Line 235: exec --no-startup-id "feh --bg-fill -z ~/wallpapers" &
ERROR: CONFIG: Line 236: exec_always --no-startup-id "~/.config/polybar/launch.sh" &
ERROR: CONFIG: Line 237: exec --no-startup-id "redshift ~/.config/redshift/redshift.conf" &
ERROR: CONFIG:                                                                            ^
ERROR: CONFIG: Line 238: 
ERROR: CONFIG: Line 239: ### FIX TOUCHPAD TOUCH-TO-CLICK ON i3 
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "foTabus down"
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "focus up"
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "focus up"
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "focus right"
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "focus right"
ERROR: Duplicate keybinding in config file:
  state mask 0x40 with keysym Tab, command "focus right"
ERROR: FYI: You are using i3 version 4.23 (2023-10-29)
ERROR: Received ConfigureNotify for unknown window 0x01200000
ERROR: Received ConfigureNotify for unknown window 0x01200000
ERROR: Received ConfigureNotify for unknown window 0x01200007
ERROR: Received ConfigureNotify for unknown window 0x01200007
ERROR: Received ConfigureNotify for unknown window 0x0120000b
ERROR: Received ConfigureNotify for unknown window 0x0120000b
ERROR: Received ConfigureNotify for unknown window 0x0120000f
ERROR: Received ExposeEvent for unknown window 0x0120000f
ERROR: Received ConfigureNotify for unknown window 0x0120000f
ERROR: Received ConfigureNotify for unknown window 0x0120000f
ERROR: Received ExposeEvent for unknown window 0x0120000f
ERROR: Received ConfigureNotify for unknown window 0x0120000f
(END)

**Comment by orestisfl** [comment]
Perhaps easier to spot the error if you include the full config

**Reply by juxtapositions-g** [reply]
sure, sorry for the long reply:
[config.txt](https://github.com/i3/i3/files/14578548/config.txt)

**Reply by orestisfl** [reply]
Not sure if it's a GitHub error but the link gives 404

**Reply by okraits** [reply]
You could also post it inline with _``._

**Reply by orestisfl** [reply, marked as answer]
To fix the errors:

- Remove the ampersands from the end of the commands
- Remove duplicate $mod+Tab bindings

```diff
diff --git a/home/orestis/Downloads/config(1).txt b/home/orestis/Downloads/config.txt
index b43a677..33e7a1a 100644
--- a/home/orestis/Downloads/config(1).txt
+++ b/home/orestis/Downloads/config.txt
@@ -103,9 +103,6 @@ bindsym $mod+d exec --no-startup-id rofi -show drun
 
 # change focus
 bindsym $mod+Tab focus left
-bindsym $mod+Tab foTabus down
-bindsym $mod+Tab focus up
-bindsym $mod+Tab focus right
 
 # alternatively, you can use the cursor keys:
 bindsym $mod+Left focus left
@@ -226,9 +223,9 @@ bindsym $mod+r mode "resize"}
 
 
 ### AUTOSTART-TINGS ###
-exec --no-startup-id "feh --bg-fill -z ~/wallpapers" &
-exec_always --no-startup-id "~/.config/polybar/launch.sh" &
-exec --no-startup-id "redshift ~/.config/redshift/redshift.conf" &
+exec --no-startup-id "feh --bg-fill -z ~/wallpapers"
+exec --no-startup-id "~/.config/polybar/launch.sh"
+exec --no-startup-id "redshift ~/.config/redshift/redshift.conf"
 
 
 ### APPS SHORTCUTS ###
```

**Reply by juxtapositions-g** [reply]
woah, thanks a lot! i already migrated to gnome after I broke my system further :DD but I'll put i3 side to side and check your answer! thanks a lot again!
